### PR TITLE
[9.0][MIG] medical insurance

### PR DESCRIPTION
**Pull request by lasley**
Upgrades medical_insurance to v9, dependent on #69

### CODE REVIEW

**Comment by lasley** [comment]
Rebased and updated to incubator code. This is ready for review.

cc @tedsalmon @kellylougheed

**Review by kellylougheed — APPROVED** [review]

**Review by tedsalmon — APPROVED** [review]
👍 LGTM!

File: medical_insurance/__init__.py [hunk 1]
```diff
@@ -1,23 +1,5 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2004-TODAY Tech-Receptives
```
Comment (jcdrubay) [inline medical_insurance/__init__.py]: Today > 201?

File: medical_insurance/__openerp__.py [hunk 2]
```diff
@@ -1,51 +1,12 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
```
Comment (jcdrubay) [inline medical_insurance/__openerp__.py]: Today > 201?

File: medical_insurance/models/__init__.py [hunk 3]
```diff
@@ -1,23 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
```
Comment (jcdrubay) [inline medical_insurance/models/__init__.py]: Today > 201?

File: medical_insurance/models/medical_insurance_company.py [hunk 4]
```diff
@@ -1,23 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
```
Comment (jcdrubay) [inline medical_insurance/models/medical_insurance_company.py]: Today > 201?

File: medical_insurance/models/medical_insurance_plan.py [hunk 5]
```diff
@@ -1,23 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
```
Comment (jcdrubay) [inline medical_insurance/models/medical_insurance_plan.py]: Today > 201?

File: medical_insurance/models/medical_insurance_template.py [hunk 6]
```diff
@@ -1,23 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
```
Comment (jcdrubay) [inline medical_insurance/models/medical_insurance_template.py]: Today > 201?

File: medical_insurance/models/medical_insurance_template.py [hunk 7]
```diff
@@ -26,10 +9,6 @@ class MedicalInsuranceTemplate(models.Model):
     _name = 'medical.insurance.template'
     _description = 'Medical Insurance Templates'
     _inherits = {'product.product': 'product_id', }
-    name = fields.Char(
```
Comment (jcdrubay) [inline medical_insurance/models/medical_insurance_template.py:29]: If we remove name, we should probably add a function `name_get` or at least set a `_rec_name`

Maybe 

```
    @api.multi
    @api.depends('product_id.name',
                 'insurance_company_id.name',
                 'plan_number')
    def name_get(self):
        result = []
        for r in self:
            name = "%s (%s,%s)" % (
                product_id.name,
                insurance_company_id.name,
                plan_number
            )
            result.append((r.id, name))
        return result
```

**Review by jcdrubay — COMMENTED** [review]
LGTM, just one comment + fix for the year of the copyright (I did not mention all of them :wink: )

File: medical_insurance/models/__init__.py (hunk 3 above)
Comment (lasley) [reply medical_insurance/models/__init__.py]: Thanks for the catches - please keep an eye out for these in the reviews. I didn't quite understand things as well as I do now regarding the copyrights in open source contributions. Live and learn, fixing past me's mistakes while we're at it 🚀

**Comment by lasley** [comment]
Copyrights fixed and name_get added. I made a few slight modifications to the proposed code, but otherwise same same. Admittedly didn't fire up the v9 local to test, so I'll double check functional for the template `name_get` once Runbot builds and I recommend a reviewer do the same 😉

**Comment by lasley** [comment]
Closing - we no longer have interest in v9. We will continue these forward in v10.

Anyone is welcome to pick the code up and reintroduce a PR, however.

**Pull request closed without merging**